test_compileall2.py

```diff
@@ -385,83 +385,161 @@ def test_ignore_symlink_destination(self):
 
     def test_hardlink_deduplication_same_bytecode_all_opt(self):
-        """Same bytecode produced for all levels of optimization"""
+        # 'a = 0' produces the same bytecode for all optimization levels
         path = os.path.join(self.directory, "test", "same_all")
         os.makedirs(path)
 
         simple_script = script_helper.make_script(path, "test_same_bytecode", "a = 0")
         pyc_opt0 = importlib.util.cache_from_source(simple_script)
-        pyc_opt1 = importlib.util.cache_from_source(simple_script, optimization=1)
-        pyc_opt2 = importlib.util.cache_from_source(simple_script, optimization=2)
+        pyc_opt1 = importlib.util.cache_from_source(
+            simple_script,
+            **compileall.optimization_kwarg(1)
+        )
+        pyc_opt2 = importlib.util.cache_from_source(
+            simple_script,
+            **compileall.optimization_kwarg(2)
+        )
 
         compileall.compile_dir(path, quiet=True, optimize=[0, 1, 2], hardlink_dupes=True)
-        self.assertTrue(os.stat(pyc_opt0).st_ino == os.stat(pyc_opt1).st_ino == os.stat(pyc_opt2).st_ino)
+
+        # import pdb; pdb.set_trace()
+
+        # All three files should have the same inode (hardlinks)
+        self.assertEqual(os.stat(pyc_opt0).st_ino, os.stat(pyc_opt1).st_ino)
+        self.assertEqual(os.stat(pyc_opt1).st_ino, os.stat(pyc_opt2).st_ino)
 
         for pyc_file in pyc_opt0, pyc_opt1, pyc_opt2:
-            os.unlink(pyc_file)
+            try:
+                os.unlink(pyc_file)
+            except OSError:
+                # Python <= 3.4 does not create all the files
+                pass
 
         compileall.compile_dir(path, quiet=True, optimize=[0, 1, 2], hardlink_dupes=False)
-        self.assertTrue(os.stat(pyc_opt0).st_ino != os.stat(pyc_opt1).st_ino != os.stat(pyc_opt2).st_ino)
+
+        # Deduplication disabled, all pyc files should have different inodes
+        self.assertNotEqual(os.stat(pyc_opt0).st_ino, os.stat(pyc_opt1).st_ino)
+        if compileall.PY35:
+            # Python 3.4 produces the same file for opt1 and opt2
+            self.assertNotEqual(os.stat(pyc_opt1).st_ino, os.stat(pyc_opt2).st_ino)
 
     def test_hardlink_deduplication_same_bytecode_some_opt(self):
+        # 'a = 0' produces the same bytecode for all optimization levels
+        # only two levels of optimization [0, 1] tested
         """Same bytecode produced for some levels of optimization"""
         path = os.path.join(self.directory, "test", "same_some")
         os.makedirs(path)
 
         simple_script = script_helper.make_script(path, "test_same_bytecode", "a = 0")
         pyc_opt0 = importlib.util.cache_from_source(simple_script)
-        pyc_opt2 = importlib.util.cache_from_source(simple_script, optimization=2)
+        pyc_opt2 = importlib.util.cache_from_source(
+            simple_script,
+            **compileall.optimization_kwarg(2)
+        )
```

Comment:
Could you please explain why changes like this were needed? It's hard to deduce.